internal/mesh/internal/types/xroute_test.go

```diff
@@ -457,19 +473,29 @@ func testXRouteACLs[R XRouteData](t *testing.T, newRoute func(t *testing.T, pare
 			reg, ok := registry.Resolve(tc.res.Id.GetType())
 			require.True(t, ok)
 
-			err = reg.ACLs.Read(authz, &acl.AuthorizerContext{}, tc.res.Id, nil)
+			authCtx := &acl.AuthorizerContext{} // unused
+
+			err = reg.ACLs.Read(authz, authCtx, tc.res.Id, nil)
 			require.ErrorIs(t, err, resource.ErrNeedData, "read hook should require the data payload")
 
-			checkF(t, "read", tc.readOK, reg.ACLs.Read(authz, &acl.AuthorizerContext{}, tc.res.Id, tc.res))
-			checkF(t, "write", tc.writeOK, reg.ACLs.Write(authz, &acl.AuthorizerContext{}, tc.res))
-			checkF(t, "list", DEFAULT, reg.ACLs.List(authz, &acl.AuthorizerContext{}))
+			checkF(t, "read", tc.readOK, reg.ACLs.Read(authz, authCtx, tc.res.Id, tc.res))
+			checkF(t, "write", tc.writeOK, reg.ACLs.Write(authz, authCtx, tc.res))
+			checkF(t, "list", DEFAULT, reg.ACLs.List(authz, authCtx))
 		})
 	}
 
-	serviceRead := func(name string) string {
+	isEnterprise := (structs.NodeEnterpriseMetaInDefaultPartition().PartitionOrEmpty() == "default")
```

Comment:
What if you are running in enterprise but in the `default` partition? In that case, we won't be testing the partitioned policy for the default partition it seems

Reply:
This specific call to `PartitionOrEmpty()` returns the following:
- CE: `""`
- ENT: `"default"`

So as long as the build was produced using `-tags consulent` then `isEnterprise` will always be `true`.

Reply:
Ah ok! My question was more about if both builds return the same value, but in this case that's not a problem.

Reply:
I'll replace this hack with https://github.com/hashicorp/consul/pull/19201 to make it more obvious what's going on.